### DISCUSSION TITLE
Ability to filter shown docker containers

### DISCUSSION THREAD

**Discussion by nil0x42** [netdata/netdata#12358] [Q&A] [answered]
Hi !
I have an issue. On my OS, i have a program who regularly launches temporary docker containers who stand some minutes only.

After some time, netdata dashboard starts accumulating hundred irrelevant docker containers.
I would like to be able to manually specify which docker container names are allowed to be shown by netdata.

It looks difficult with cgroups, because on /sys/fs/cgroup/*/docker/*, names are a long alphanumeric string who may change over time (when an image is upgraded).
So i would like to be able to e.g, only allow `myDocker1` & `myDocker2` names to be collected, or at least shown in the dashboard.

Is that possible ?

**Comment by ilyam8** [comment]
Hi, @nil0x42.

> temporary docker containers who stand some minutes only
> netdata dashboard starts accumulating hundred irrelevant docker containers.

What dashboard are you referring to? The Cloud dashboard or Agent(ip:19999) dashboard? All deleted/stopped containers should disappear from the Agent dashboard (up to 10 seconds).

**Reply by nil0x42** [reply]
Well, i am using app.netdata.cloud, but all containers who have existed at some point in the selected timerange do show up
And i understand that it might be a good default, but i would like very nuch to be able to filter-out by container name (regex/pattern)

**Reply by ilyam8** [reply]
> app.netdata.cloud
> netdata dashboard starts accumulating hundred irrelevant docker containers

This appears to be a [bug](https://github.com/netdata/netdata/issues/12443). 

---

> only allow myDocker1 & myDocker2 names to be collected

DO you want them to be **not collected** or to be able to filter out them from the dashboard (**not shown**)?

**Reply by nil0x42** [reply]
Ideally, i would want to not collect, but if it isn't possible, i would like to at least hide them from dashboard.

**Reply by ilyam8** [reply]
If I understand you correctly, adding a new configuration option will do - in addition to `enable by default cgroups matching` to add `enable by default cgroups names matching`. If so, I suggest [opening a feature request](https://github.com/netdata/netdata/issues/new?assignees=&labels=feature+request%2Cneeds+triage&template=FEAT_REQUEST.yml&title=%5BFeat%5D%3A+).

**Comment by ilyam8** [comment, marked as answer]
Created a [feature request ](https://github.com/netdata/netdata/issues/12564). @nil0x42 you can subscribe to that issue to receive updates.

**Reply by ilyam8** [reply]
implemented in https://github.com/netdata/netdata/pull/12778

```cmd
enable by default cgroups names matching =  !nginx2 *
```
